packages/graphql/src/translate/queryAST/ast/operations/composite/CompositeAggregationOperation.ts

```diff
@@ -0,0 +1,214 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Cypher from "@neo4j/cypher-builder";
+import type { InterfaceEntityAdapter } from "../../../../../schema-model/entity/model-adapters/InterfaceEntityAdapter";
+import type { UnionEntityAdapter } from "../../../../../schema-model/entity/model-adapters/UnionEntityAdapter";
+import { filterTruthy } from "../../../../../utils/utils";
+import type { QueryASTContext } from "../../QueryASTContext";
+import type { QueryASTNode } from "../../QueryASTNode";
+import type { AggregationField } from "../../fields/aggregation-fields/AggregationField";
+import type { Filter } from "../../filters/Filter";
+import type { AuthorizationFilters } from "../../filters/authorization-filters/AuthorizationFilters";
+import type { Pagination } from "../../pagination/Pagination";
+import type { Sort, SortField } from "../../sort/Sort";
+import type { OperationTranspileOptions, OperationTranspileResult } from "../operations";
+import { Operation } from "../operations";
+import type { CompositeAggregationPartial } from "./CompositeAggregationPartial";
+
+export class CompositeAggregationOperation extends Operation {
+    private children: CompositeAggregationPartial[];
+    protected directed: boolean;
+    public fields: AggregationField[] = []; // Aggregation fields
+    public nodeFields: AggregationField[] = []; // Aggregation node fields
+    public edgeFields: AggregationField[] = []; // Aggregation node fields
+
+    private entity: InterfaceEntityAdapter | UnionEntityAdapter;
+
+    protected authFilters: AuthorizationFilters[] = [];
+
+    protected filters: Filter[] = [];
+    protected pagination: Pagination | undefined;
+    protected sortFields: Sort[] = [];
+
+    public nodeAlias: string | undefined; // This is just to maintain naming with the old way (this), remove after refactor
+
+    constructor({
+        compositeEntity,
+        children,
+        directed = true,
+    }: {
+        compositeEntity: InterfaceEntityAdapter | UnionEntityAdapter;
+        children: CompositeAggregationPartial[];
+        directed?: boolean;
+    }) {
+        super();
+        this.entity = compositeEntity;
+        this.children = children;
+        this.directed = directed;
+    }
+
+    public getChildren(): QueryASTNode[] {
+        return filterTruthy([
+            ...this.fields,
+            ...this.nodeFields,
+            ...this.edgeFields,
+            ...this.filters,
+            ...this.sortFields,
+            ...this.authFilters,
+            this.pagination,
+            ...this.children,
+        ]);
+    }
+
+    protected getSortFields(context: QueryASTContext, target: Cypher.Variable): SortField[] {
+        return this.sortFields.flatMap((sf) => sf.getSortFields(context, target, false));
+    }
+
+    public transpile({ context }: OperationTranspileOptions): OperationTranspileResult {
```

Comment:
We may want to consider moving some parts of this into separate private methods. For instance the fieldSubqueries calculation